research-software-directory/RSD-as-a-service
# Add possibility to define a whitelist for user logins

# User Whitelist

See https://github.com/research-software-directory/Helmholtz/issues/4

This helps us for a first productive deployment of the RSD in order to evaluate, present, test and get further feedback.
So that only some defined test users can login to the RSD and create entries and we not ending up having many "test 123" entries in the list :wink: 

Changes proposed in this pull request:

* add configuration variable to define user mail addresses which are allowed to access the RSD
* this should only be considered as a temporary implementation as long as we have no role management
* whitelist is **not** activated for the local login

How to test:

* set the env variable `RSD_AUTH_USER_MAIL_WHITELIST` to the mail address of your login account
* start the containers, you should be able to login only with this account
* unset the env variable again and every user should login again
* you can also define multiple mail addresses with a comma-separated list in the env value

PR Checklist:
